### ISSUE TITLE
Postgres throws 'invalid snapshot identifier' exception when exporting data from large tables

### ISSUE THREAD

**Issue by tomsanbear** [duckdb/postgres_scanner#154]
### What happens?

When running a COPY command from a large postgres table (27025183), an error is raised.

`Error: Invalid Error: Failed to execute query "SET TRANSACTION SNAPSHOT '00000004-000002BE-1'": ERROR:  invalid snapshot identifier: "00000004-000002BE-1"`

This does not get raised with smaller tables of hundreds/thousands of rows.

The postgres container is deployed in a docker container, if that is relevant to any configuration/debugging steps. Logs from the postgres container:

```
postgres-1  | 2023-12-20 18:58:12.535 UTC [693] ERROR:  invalid snapshot identifier: "00000004-0000021D-1"
```

### To Reproduce

```
COPY (
  SELECT
    *
  FROM
    table_name
) TO 'data/table_name.parquet' (FORMAT PARQUET);
```

### OS:

Ubuntu Linux

### PostgreSQL Version:

15

### DuckDB Version:

0.9.2

### DuckDB Client:

Linux CLI

### Full Name:

Thomas Santerre

### Affiliation:

N/A

### Have you tried this on the latest `main` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by tomsanbear** [comment]
Note I have attempted the suggestions from a previous issue:

```
force install postgres_scanner
from
  'http://nightly-extensions.duckdb.org';

SET
  pg_connection_limit = 1;

CALL postgres_attach (
  'host=127.0.0.1 port=5432 dbname=postgres user=postgres password=password'
);

COPY (
  SELECT
    *
  FROM
    table_name
) TO 'data/table_name.parquet' (FORMAT PARQUET);
```

Which results in the same error

**Comment by Mytherin** [comment]
Thanks for the report! Do you run into the same issue with the new [`ATTACH` syntax](https://github.com/duckdb/postgres_scanner#reading-data-from-postgres)? `postgres_attach` is the old syntax and is deprecated.

**Comment by davetapley** [comment]
@Mytherin seeing same here. I tried new `ATTACH`, but ran in to #156 😑

**Comment by hazelgurule** [comment]
I'm on Postgres 14.10, and this is happening both on the Python and Node duckdb client. 

Also seeing these `INVALID SNAPSHOT IDENTIFIER` errors when I run `POSTGRES_SCAN_PUSHDOWN()` without the new `ATTACH` syntax. 

This didn't happen when I first started using duckdb-postgres 0.9.2 in December, but I cleared out my local .duckdb extensions cache today, and the queries that were previously working are now failing.

When I switch to `ATTACH` syntax, I get an error like https://github.com/duckdb/postgres_scanner/issues/156 `IO Error: Unable to read binary COPY data from Postgres: no COPY in progress`, and when I set `SET threads=1` it starts to work again.

**Comment by ben-pr-p** [comment]
I'm also getting this - for me, downgrading to 0.9.1 fixed it on my M2 Mac. Will try it out in a Linux container later.

**Comment by Mytherin** [comment]
I've pushed a fix for this in #164. Note that this is only an issue for the legacy `postgres_scan` functions and in general we recommend moving away from those functions to the new `ATTACH` syntax.

**Comment by iangow** [comment]
This seems to have broken `read_postgres()` with DuckDB backend and Ibis. See comment just above.
